backends/postgres/postgres_backend_test.go

```diff
@@ -773,3 +774,176 @@ func Test_ConnectionTimeout(t *testing.T) {
+			want:    "postgres://username:password@hostname:5432/database?sslmode=verify-full&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:  "encoded password is preserved",
+			input: "postgres://username:pass%21%40%23$%25%5E&%2A%28%29%3A%2F%3Fword@hostname:5432/database",
+			want: fmt.Sprintf(
+				"postgres://%s@hostname:5432/database?sslmode=require&x-migrations-table=neoq_schema_migrations",
+				url.UserPassword("username", "pass!@#$%^&*():/?word").String(),
+			),
+			wantErr: false,
+		},
+		{
+			name:    "multiple hostnames",
+			input:   "postgres://username:password@hostname1,hostname2,hostname3:5432/database",
+			want:    "postgres://username:password@hostname1,hostname2,hostname3:5432/database?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+
+		// Examples connstrings from https://www.postgresql.org/docs/16/libpq-connect.html
+		{
+			name:    "valid empty postgresql scheme input",
+			input:   "postgresql://",
+			want:    "postgresql:?sslmode=disable&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "hostname localhost",
+			input:   "postgresql://localhost",
+			want:    "postgresql://localhost?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "hostname localhost with custom port",
+			input:   "postgresql://localhost:5433",
+			want:    "postgresql://localhost:5433?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "non-default database",
+			input:   "postgresql://localhost/mydb",
+			want:    "postgresql://localhost/mydb?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "username",
+			input:   "postgresql://user@localhost",
+			want:    "postgresql://user@localhost?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "username and password",
+			input:   "postgresql://user:secret@localhost",
+			want:    "postgresql://user:secret@localhost?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "custom params are ignored",
+			input:   "postgresql://other@localhost/otherdb?connect_timeout=10&application_name=myapp",
+			want:    "postgresql://other@localhost/otherdb?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+		{
+			name:    "multiple hostnames and ports",
+			input:   "postgresql://host1:123,host2:456/somedb?target_session_attrs=any&application_name=myapp",
+			want:    "postgresql://host1:123,host2:456/somedb?sslmode=require&x-migrations-table=neoq_schema_migrations",
+			wantErr: false,
+		},
+
+		// Inputs that cause errors
+		{
+			name:    "pq-style input returns error",
+			input:   "host=localhost port=5432 dbname=mydb connect_timeout=10",
+			want:    "",
+			wantErr: true,
+		},
```

Comment:
@acaloiaro pq-style strings are expected to cause an error with the current implementation in my PR. Please let me know if you'd like me to add support for them

Reply:
I would prefer pq-style connection strings to be supported if it's not too much trouble.

Reply:
hi @acaloiaro I apologize about the delay, I just pushed up commit c5d1bfcabe75cf79918f0393e14145333e00416a which adds support for pq-style strings.

The input string will still go through `pgx.ParseConfig` which will make sure it was in a valid format before it's returned unmodified.

Let me know what you think